microcosm_postgres/migrate.py

```diff
@@ -0,0 +1,203 @@
+"""
+Alembic CLI with simplified configuration.
+
+Alembic is a terrific tool that makes some unfortunate choices about configuration,
+expecting a verbose directory structure with several layers of configuration.
+
+This module monkey patches Alembic's CLI tool to work better within a microcosm.
+
+To use this entry enty point instead of the Alembic CLI:
+
+ 0. Don't use `alembic init`
+
+ 1. Write models that extend `microcosm_postgres.models.Model`
+
+ 2. Add a `migrations` directory within your source tree.
+
+    This directory does not need to be an importable Python module, but it should
+    be included as part of your distribution so that migrations ship with the service.
+
+ 3. Initialize your object graph (including your models) and ensure that it knows the
+    migrations directory path.
+
+    For example:
+
+        from microcosm_postgres.example import Company, Employee  # noqa
+        graph = create_object_graph(name="example", testing=True)
+        graph.migrations_dir = join(dirname(__file__), "migrations")
```

Comment:
I'm not a big fan of having to tell the system where to find migrations. I'd prefer to have a convention of where to look for migrations, but couldn't come up with anything I liked. (It's also sloppy the way I just attached this value to the graph, instead of using the config object and a factory. I intend to fix that, but wanted to discuss the approach first.)

The closest I came to defining a convention was to rely on the graph's name as a key into `sys.modules` and derive a migrations path from the load location of the module with that name. That probably works for concrete services (which are likely to use their own name), but fails in the example/testing cases I was using (where I chose the name "example" for clarity, but the top-level module is actually `microcosm_postgres`).